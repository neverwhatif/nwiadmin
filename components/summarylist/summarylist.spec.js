import React from 'react';
import { shallow } from 'enzyme';

import SummaryList from './index';

jest.mock('./summarylistitem', () => 'SummaryListItem');

const data = {
    Lorem: 'Lorem',
    Ipsum: 'Ipsum',
};

describe('SummaryList', () => {
    const component = shallow(<SummaryList data={data} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render list items', () => {
        expect(component.find('SummaryListItem').length).toBe(2);
    });
});
