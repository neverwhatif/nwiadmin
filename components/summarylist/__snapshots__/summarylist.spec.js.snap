// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryList should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SummaryList
    data={
      Object {
        "Ipsum": "Ipsum",
        "Lorem": "Lorem",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SummaryListItem
          def="Lorem"
          term="Lorem"
        />,
        <SummaryListItem
          def="Ipsum"
          term="Ipsum"
        />,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Lorem",
        "nodeType": "host",
        "props": Object {
          "def": "Lorem",
          "term": "Lorem",
        },
        "ref": null,
        "rendered": null,
        "type": "SummaryListItem",
      },
      Object {
        "instance": null,
        "key": "Ipsum",
        "nodeType": "host",
        "props": Object {
          "def": "Ipsum",
          "term": "Ipsum",
        },
        "ref": null,
        "rendered": null,
        "type": "SummaryListItem",
      },
    ],
    "type": "dl",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SummaryListItem
            def="Lorem"
            term="Lorem"
          />,
          <SummaryListItem
            def="Ipsum"
            term="Ipsum"
          />,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Lorem",
          "nodeType": "host",
          "props": Object {
            "def": "Lorem",
            "term": "Lorem",
          },
          "ref": null,
          "rendered": null,
          "type": "SummaryListItem",
        },
        Object {
          "instance": null,
          "key": "Ipsum",
          "nodeType": "host",
          "props": Object {
            "def": "Ipsum",
            "term": "Ipsum",
          },
          "ref": null,
          "rendered": null,
          "type": "SummaryListItem",
        },
      ],
      "type": "dl",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
